kubernetes/samples/scripts/domain-lifecycle/helper.sh

```diff
@@ -0,0 +1,605 @@
+# $1 - Current number of replicas
+# $2 - Server start policy
+# $3 - Replica count
+# $4 - Returns "true" or "false" indicating if server should start.
+#
+function shouldStart {
+  local currentReplicas=$1
+  local policy=$2
+  local replicaCount=$3 
+  local __result=$4
+
+  if [ "$policy" == "ALWAYS" ]; then
+    eval $__result=true
+  elif [ "$policy" == "NEVER" ]; then
+    eval $__result=false
+  elif [ "${currentReplicas}" -lt "${replicaCount}" ]; then
+    eval $__result=true
+  else 
+    eval $__result=false
+  fi
+}
+
+#
+# Function to create patch string for updating replica count
+# $1 - Domain resource in json format
+# $2 - Name of cluster whose replica count will be patched
+# $3 - operation string indicating whether to increment or decrement replica count. 
+#      Valid values are "INCREMENT" and "DECREMENT"
+# $4 - Return value containing replica update patch string
+# $5 - Return value containing updated replica count
+#
+function createReplicaPatch {
+  local domainJson=$1
+  local clusterName=$2
+  local operation=$3
+  local __result=$4
+  local __replicaCount=$5
+  local maxReplicas=""
+  local infoMessage="Current replica count value is same as maximum number of replica count. \
+Not increasing replica count value."
+
+  maxReplicaCmd="(.status.clusters[] | select (.clusterName == \"${clusterName}\")) \
+    | .maximumReplicas"
+  getReplicaCount  "${domainJson}" "${clusterName}" replica
+  if [ "${operation}" == "DECREMENT" ]; then
+    replica=$((replica-1))
+    if [ ${replica} -lt 0 ]; then
+      replica=0
+    fi
+  elif [ "${operation}" == "INCREMENT" ]; then
+    maxReplicas=$(echo ${domainJson} | jq "${maxReplicaCmd}")
+    if [ ${replica} -eq ${maxReplicas} ]; then
+      printInfo "${infoMessage}"
+    else
+      replica=$((replica+1))
+    fi
+  fi
+
+  cmd="(.spec.clusters[] | select (.clusterName == \"${clusterName}\") \
+    | .replicas) |= ${replica}"
+  replicaPatch=$(echo ${domainJson} | jq "${cmd}" | jq -cr '(.spec.clusters)')
+  eval $__result="'${replicaPatch}'"
+  eval $__replicaCount="'${replica}'"
+}
+
+#
+# Function to validate whether a server belongs to a  cluster or is an independent managed server
+# $1 - Domain unique id.
+# $2 - Domain namespace.
+# $3 - Server name.
+# $4 - Return value of "true" or "false" indicating if server is valid (i.e. if it's part of a cluster or independent server).
+# $5 - Retrun value containting cluster name to which this server belongs.
+#
+function validateServerAndFindCluster {
+  local domainUid=$1
+  local domainNamespace=$2 
+  local serverName=$3
+  local __isValidServer=$4
+  local __clusterName=$5
+  local serverCount=""
+  local errorMessage="Server name is outside the range of allowed servers. \
+Please make sure server name is correct."
+
+  eval $__isValidServer=false
+  eval $__clusterName=UNKNOWN
+  configMap=$(${kubernetesCli} get cm ${domainUid}-weblogic-domain-introspect-cm \
```

Comment:
Right now, the script will simply fail-fast if the cm isn't found (due to the global 'set -e').  I suspect it'd be more helpful to the user if it generated a clear error message...  If you want to address this, you can use the same pattern as was used in 'timestamp':  `val=$(set +e && dostuff || echo error)`